server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -183,6 +191,45 @@ public Invocations getInvocations() {
       return CompletableFuture.completedFuture(responseBuilder.build());
    }
 
+   private CompletionStage<RestResponse> getAllDetails(RestRequest request) {
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      String cacheName = request.variables().get("cacheName");
+
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, APPLICATION_JSON, APPLICATION_JSON, request);
+      if (cache == null) {
+         responseBuilder.status(HttpResponseStatus.NOT_FOUND);
+         return CompletableFuture.completedFuture(responseBuilder.build());
+      }
+
+      Stats stats = cache.getAdvancedCache().getStats();
+      Configuration configuration = cache.getCacheConfiguration();
+      int size = cache.getAdvancedCache().size();
+      boolean rehashInProgress = cache.getAdvancedCache().getDistributionManager().isRehashInProgress();
+      String version = cache.getVersion();
+      LocalizedCacheTopology cacheTopology = cache.getAdvancedCache().getDistributionManager().getCacheTopology();
+      Set<Address> membersSet = cacheTopology.getMembersSet();
+      ConsistentHash currentCH = cacheTopology.getCurrentCH();
```

Comment:
I am not 100% sure, but I think one of those methods will throw a NPE if the cache is ```local```. Better add a test to check :)

Reply:
ok